homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -0,0 +1,223 @@
+            return conn
+
+        def get_router_title(conn: Connection) -> str:
+            """Get title for router."""
+            title = None
+            client = Client(conn)
+            try:
+                info = client.device.basic_information()
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.debug("Could not get device.basic_information", exc_info=True)
+            else:
+                title = info.get("devicename")
+            if not title:
+                try:
+                    info = client.device.information()
+                except Exception:  # pylint: disable=broad-except
+                    _LOGGER.debug("Could not get device.information", exc_info=True)
+                else:
+                    title = info.get("DeviceName")
+            return title or DEFAULT_DEVICE_NAME
+
+        username = user_input.get(CONF_USERNAME)
+        password = user_input.get(CONF_PASSWORD)
+        try:
+            conn = await self.hass.async_add_executor_job(
+                try_connect, username, password
+            )
+        except LoginErrorUsernameWrongException:
+            errors[CONF_USERNAME] = "incorrect_username"
+        except LoginErrorPasswordWrongException:
+            errors[CONF_PASSWORD] = "incorrect_password"
+        except LoginErrorUsernamePasswordWrongException:
+            errors[CONF_USERNAME] = "incorrect_username_or_password"
+        except LoginErrorUsernamePasswordOverrunException:
+            errors["base"] = "login_attempts_exceeded"
+        except ResponseErrorException:
+            _LOGGER.warning("Response error", exc_info=True)
+            errors["base"] = "response_error"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.warning("Unknown error connecting to device", exc_info=True)
+            errors[CONF_URL] = "unknown_connection_error"
+        if errors:
+            await self.hass.async_add_executor_job(logout)
+            return await self._async_show_user_form(
+                user_input=user_input, errors=errors
+            )
+
+        title = await self.hass.async_add_executor_job(get_router_title, conn)
+        await self.hass.async_add_executor_job(logout)
+
+        return self.async_create_entry(title=title, data=user_input)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Huawei LTE options flow."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        data_schema = vol.Schema(
+            OrderedDict(
+                (
+                    (
+                        vol.Optional(
+                            CONF_TRACK_NEW,
```

Comment:
@balloob do we want each device tracker platform to implement this option?

It will be a lot of repetition, since probably all device scanners need this.

Reply:
We should remove this option. This is the same as the system option https://developers.home-assistant.io/docs/en/entity_registry_disabled_by.html#config-entry-system-options-setting-default-value-of-disabled_by-for-new-entity-registry-entries

Reply:
The system option affects the entire router, whereas this option is just for the device tracker. Do you want it removed anyway?

Reply:
Also, I would like to be able to not poll the associated device list if the user does not want anything from there. If I remove this option, do you think we should disable the poll if new entities are disabled?

Otherwise, we would have no option but to always enable the poll and accumulate disabled device tracker entities in the entity registry even if the user is never going to do anything with them, and the user could not do anything about that. That's something I'd like to avoid.

Reply:
The idea behind still filling up the entity registry is that a user can decide to enable specific entities that it wants to follow. It would be up to the integration to see if any entity needs updating (based on a callback to enable polling in `async_added_to_hass` callback) and only poll if necessary.

Since users will only be able to set this option after they create the integration, the sensors should already been added ?